ui-v2/app/components/feedback-dialog.js

```diff
@@ -1,42 +1,49 @@
 import Component from '@ember/component';
 import { get, set } from '@ember/object';
 import { inject as service } from '@ember/service';
+import qsaFactory from 'consul-ui/utils/qsa-factory';
+const $$ = qsaFactory();
 
 import SlotsMixin from 'ember-block-slots';
 const STATE_READY = 'ready';
 const STATE_SUCCESS = 'success';
 const STATE_ERROR = 'error';
 export default Component.extend(SlotsMixin, {
   wait: service('timeout'),
-  interval: null,
   classNames: ['with-feedback'],
+  transition: '',
+  transitionClassName: 'feedback-dialog-out',
   state: STATE_READY,
   permanent: true,
   init: function() {
     this._super(...arguments);
     this.success = this._success.bind(this);
     this.error = this._error.bind(this);
   },
-  _success: function() {
-    set(this, 'state', STATE_SUCCESS);
-    get(this, 'wait')
-      .execute(3000, interval => {
-        clearInterval(get(this, 'interval'));
-        set(this, 'interval', interval);
+  applyTransition: function() {
+    const wait = get(this, 'wait').execute;
+    const className = get(this, 'transitionClassName');
+    wait(0)
+      .then(() => {
+        set(this, 'transition', className);
+        return wait(0);
       })
       .then(() => {
+        return new Promise(resolve => {
+          $$(`.${className}`, this.element)[0].addEventListener('transitionend', resolve);
+        });
+      })
+      .then(() => {
+        set(this, 'transition', '');
         set(this, 'state', STATE_READY);
       });
   },
```

Comment:
I tried for ages to get a good `nextTick` using the ember runloop things. Nothing would give me a reliable 'do this after render' though. Went with the old `setTimeout(function(){}, 0)` in the end, seeing as I was using similar previously anyway. I'm thinking of changing that `wait` service to accept no arguments with a default of `0` so I can write things like:

```javascript
const tick = get(this, 'wait').execute;
tick().then(
  function() {
    // do something
    return tick()
}).then(
  function() {
    // do something else
  }
)
```